application/src/main/java/org/thingsboard/server/service/cloud/PostgresUplinkMessageService.java

```diff
@@ -0,0 +1,221 @@
+import com.drew.lang.annotations.Nullable;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.thingsboard.server.common.data.AttributeScope;
+import org.thingsboard.server.common.data.cloud.CloudEvent;
+import org.thingsboard.server.common.data.id.TenantId;
+import org.thingsboard.server.common.data.kv.AttributeKvEntry;
+import org.thingsboard.server.common.data.kv.BaseAttributeKvEntry;
+import org.thingsboard.server.common.data.kv.LongDataEntry;
+import org.thingsboard.server.common.data.page.PageData;
+import org.thingsboard.server.common.data.page.TimePageLink;
+import org.thingsboard.server.dao.attributes.AttributesService;
+import org.thingsboard.server.dao.cloud.CloudEventService;
+import org.thingsboard.server.service.cloud.rpc.CloudEventStorageSettings;
+import org.thingsboard.server.service.executors.DbCallbackExecutorService;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import static org.thingsboard.server.service.cloud.MessageConstants.FAILED_TO_UPDATE_QUEUE_OFFSET_ERROR_MESSAGE;
+import static org.thingsboard.server.service.cloud.MessageConstants.QUEUE_OFFSET_WAS_UPDATED_MESSAGE;
+import static org.thingsboard.server.service.cloud.MessageConstants.FAILED_TO_FIND_CLOUD_EVENTS;
+import static org.thingsboard.server.service.cloud.MessageConstants.STARTED_NEW_CYCLE_MESSAGE;
+import static org.thingsboard.server.service.cloud.MessageConstants.QUEUE_END_TS_LESS_THAN_CURRENT_TIME_MESSAGE;
+import static org.thingsboard.server.service.cloud.MessageConstants.TABLE_STARTED_NEW_CYCLE_MESSAGE;
+import static org.thingsboard.server.service.cloud.MessageConstants.UPDATE_QUEUE_START_TS_SEQ_ID_OFFSET_MESSAGE;
+
+
+@Slf4j
+public abstract class PostgresUplinkMessageService extends BaseUplinkMessageService {
+
+    @Autowired
+    private CloudEventStorageSettings cloudEventStorageSettings;
+
+    @Autowired
+    private AttributesService attributesService;
+
+    @Autowired
+    private DbCallbackExecutorService dbCallbackExecutorService;
+
+    @Autowired
+    protected CloudEventService cloudEventService;
+
+    public void processHandleMessages(TenantId tenantId) throws Exception {
+        Long cloudEventsQueueSeqIdStart = getQueueSeqIdStart(tenantId).get();
+        TimePageLink cloudEventsPageLink = newCloudEventsAvailable(tenantId, cloudEventsQueueSeqIdStart);
+
+        if (cloudEventsPageLink != null) {
+            processCloudEvents(tenantId, cloudEventsQueueSeqIdStart, cloudEventsPageLink);
+        }
+    }
+
+    @Nullable
+    public TimePageLink newCloudEventsAvailable(TenantId tenantId, Long queueSeqIdStart) {
+        try {
+            return tryFindNewTimePageLink(tenantId, queueSeqIdStart);
```

Comment:
Don't like the idea of "try" in the name of the method, as we discussed. Also no need to split logic in separate method with 1 usage